Set _KDE_NET_WM_DESKTOP_FILE x11 window property.

pchote opened the pull request:
This PR fixes a small polish annoyance when running the appimages with https://github.com/TheAssassin/AppImageLauncher integrations. Different components in the KDE desktop seem to have different heuristics for matching windows to their icons, and this was the only way I could find to properly solve all of them.

I suggest reviewing [with ?w=1](https://github.com/OpenRA/OpenRA/pull/19553/files?w=1) to get a clearer picture of what has actually changed.

Before:
<img width="392" alt="Screenshot 2021-07-21 at 19 20 02" src="https://user-images.githubusercontent.com/167819/126539660-d86ed74d-f612-46cd-9e48-dec7495949af.png">

After:
<img width="395" alt="Screenshot 2021-07-21 at 19 17 34" src="https://user-images.githubusercontent.com/167819/126539680-65732687-4bb9-4ed8-a2d5-62d10dca98d7.png">

Mailaender commented on OpenRA.Game/Game.cs:
```diff
@@ -363,6 +363,10 @@ static void Initialize(Arguments args)
 				}
 			}
 
+			var kdeDesktopFile = args.GetValue("Engine.KDEDesktopFile", null);
```
This shouldn't be part of the game engine. Put it in `OpenRA.LinuxLauncher` or similar.

pchote replied:
I wouldn't object on principle to introducing a new launcher project, but how would that actually work? We need to know the X11 window ID before we can set this, which is an internal implementation detail of the platform window, which is owned by the engine. The only other way I can think of to set this without the engine knowing would be to use a helper thread and a pile of reflection to dig around in the internal state, which would be fragile and prone to race conditions. Am I missing something obvious here?

Mailaender replied:
Get rid of the command line argument here and check for `$XDG_CURRENT_DESKTOP` to detect KDE instead.

pchote replied:
Right, using environment variables is indeed something obvious that I was missing 🤦

I will rewrite this to have the platform window look for an `OPENRA_DESKTOP_FILE` environment variable, following the precident of our existing `OPENRA_DISPLAY_SCALE`. I will then use an `XDG_CURRENT_DESKTOP == KDE` check to only define the X11 property when actually needed.

pchote committed:
Set _KDE_NET_WM_DESKTOP_FILE x11 window property.

This explicitly tells KDE to associate the OpenRA window with
the integrated appimage desktop file, allowing it to use the
correct resolution icon for the task switcher.

pchote commented:
Updated.

abcdefg30 reviewed (approved):
Didn't spot any regressions on Windows or any issues glancing over the code. I trust your tests that this works as intended on non-Windows.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/723dfa4dcaf3f67828b939072aca5b10fd91bfad...819443e67bea3b10a0b851bf7c4f41c1a53ebcf6)

Mailaender commented:
This doesn't work in the AppImage

Mailaender commented:
This doesn't work in the AppImage:

![image](https://user-images.githubusercontent.com/756669/134770587-3ab63b89-ded0-4c1c-8aad-aedecb1c8d0c.png)

or it requires changes to the SDK scripts.

pchote commented:
It does require changes to the SDK scripts.

pchote commented:
It works fine for me, but of course you do need host integration (e.g. through AppImageLauncher) activated for this and any other icons/launchers/mime types to work.

Mailaender commented:
Okay, in that case https://github.com/OpenRA/OpenRA/pull/19528 was what I really wanted. I have never heard of https://github.com/TheAssassin/AppImageLauncher and I doubt many people use it.

pchote commented:
Yes, the desktop integration situation for AppImages is a real mess. They still state that individual applications managing their own integrations violates best practices, but provide no standard solution that we can expect users to willingly install.

See https://github.com/AppImage/AppImageKit/issues/1036 and related discussions for more information.
